src/github.com/getlantern/lantern-android/client/client.go

```diff
@@ -0,0 +1,209 @@
+package client
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"strconv"
+
+	"github.com/getlantern/balancer"
+)
+
+const (
+	httpConnectMethod = "CONNECT"
+	xFlashlightQOS    = "X-Flashlight-QOS"
+)
+
+// clientConfig holds global configuration settings for all clients.
+var clientConfig *Config
+
+// init attempts to setup client configuration.
+func init() {
+	var err error
+	if clientConfig, err = getConfig(); err != nil {
+		// getConfig() guarantees to return a *Config struct, so we can log the
+		// error without stopping the program.
+		log.Printf("Error updating configuration over the network: %q.", err)
+	}
+}
+
+// Client is a HTTP proxy that accepts connections from local programs and
+// proxies these via remote flashlight servers.
+type Client struct {
+	Addr string
+
+	frontedServers []*frontedServer
+	ln             *Listener
+
+	rpCh          chan *httputil.ReverseProxy
+	rpInitialized bool
+
+	balInitialized bool
+	balCh          chan *balancer.Balancer
+}
+
+// AddFrontedServer adds a fronted server to the list.
+func (client *Client) AddFrontedServer(fs *frontedServer) error {
```

Comment:
The `Client` type must be exported so it can be used in the library bindings: https://github.com/getlantern/lantern-android/blob/master/libflashlight/bindings/main.go, but I see your point and certainly other types this package is exporting such as `CA`, `Config` and `Listener` aren't mean to be used outside the client so I'd better fix them.